ts/Core/Axis/Axis.ts

```diff
@@ -1905,6 +1905,7 @@ class Axis {
             options.minTickInterval,
             // In datetime axes, don't go below the data interval, except when
             // there are scatter-like series involved (#13369).
+            !axis.options.grid &&
```

Comment:
Instead of creating an exception for the grid axis in the core, can we just set the default `minTickInterval` to 0 or 1 in the [grid axis defaults](https://github.com/highcharts/highcharts/blob/815d2def69770e22827569c6d5a65cd606390863/ts/Core/Axis/GridAxis.ts#L173-L200)?

It seems to work as intended when setting the option explicitly: https://jsfiddle.net/2r83tpuo/